### ISSUE TITLE
Can't start Docker container with INITIAL_API_KEY env variable 

### ISSUE THREAD

**Issue by bbednarek** [shlinkio/shlink#1866]
### Shlink version

3.6.3

### PHP version

All

### How do you serve Shlink

Other (explain in summary)

### Database engine

SQLite

### Database version

N/A

### Current behavior

When I supply `INITIAL_API_KEY` env variable I get following error:
```
handle_serve_command: Function call error:
	got initial serve error from the Vertex *http.Plugin, stopping execution, error: static_pool_allocate_workers: WorkerAllocate:
	goridge_frame_receive: validation failed on the message sent to STDOUT, see: https://roadrunner.dev/docs/known-issues-stdout-crc/2.x/en, invalid message:
Fatal error: Uncaught PDOException: SQLSTATE[HY000]: General error: 5 database is locked in /etc/shlink/vendor/doctrine/dbal/src/Driver/PDO/Statement.php:121
Stack trace:
#0 /etc/shlink/vendor/doctrine/dbal/src/Driver/PDO/Statement.php(121): PDOStatement->execute(NULL)
#1 /etc/shlink/vendor/doctrine/dbal/src/Statement.php(190): Doctrine\DBAL\Driver\PDO\Statement->execute(NULL)
#2 /etc/shlink/vendor/doctrine/dbal/src/Statement.php(249): Doctrine\DBAL\Statement->execute(NULL)
#3 /etc/shlink/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php(284): Doctrine\DBAL\Statement->executeStatement()
#4 /etc/shlink/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php(1144): Doctrine\ORM\Persisters\Entity\BasicEntityPersister->executeInserts()
#5 /etc/shlink/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php(430): Doctrine\ORM\UnitOfWork->executeInserts(Object(Doctrine\ORM\Mapping\ClassMetadata))
#6 /etc/shlink/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php(403): Doctrine\ORM\UnitOfWork->commit(NULL)
#7 /etc/shlink/module/Rest/src/ApiKey/Repository/ApiKeyRepository.php(28): Doctrine\ORM\EntityManager->flush()
#8 /etc/shlink/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php(272): Shlinkio\Shlink\Rest\ApiKey\Repository\ApiKeyRepository->Shlinkio\Shlink\Rest\ApiKey\Repository\{closure}(Object(Doctrine\ORM\EntityManager))
#9 /etc/shlink/module/Rest/src/ApiKey/Repository/ApiKeyRepository.php(16): Doctrine\ORM\EntityManager->wrapInTransaction(Object(Closure))
#10 /etc/shlink/module/Rest/src/ApiKey/InitialApiKeyDelegator.php(29): Shlinkio\Shlink\Rest\ApiKey\Repository\ApiKeyRepository->createInitialApiKey('test_api_key')
#11 /etc/shlink/module/Rest/src/ApiKey/InitialApiKeyDelegator.php(19): Shlinkio\Shlink\Rest\ApiKey\InitialApiKeyDelegator->createInitialApiKey('test_api_key', Object(Laminas\ServiceManager\ServiceManager))
#12 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(620): Shlinkio\Shlink\Rest\ApiKey\InitialApiKeyDelegator->__invoke(Object(Laminas\ServiceManager\ServiceManager), 'Mezzio\\Applicat...', Object(Closure), NULL)
#13 /etc/shlink/vendor/mezzio/mezzio/src/Container/ApplicationConfigInjectionDelegator.php(55): Laminas\ServiceManager\ServiceManager::Laminas\ServiceManager\{closure}()
#14 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(620): Mezzio\Container\ApplicationConfigInjectionDelegator->__invoke(Object(Laminas\ServiceManager\ServiceManager), 'Mezzio\\Applicat...', Object(Closure), NULL)
#15 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(623): Laminas\ServiceManager\ServiceManager::Laminas\ServiceManager\{closure}()
#16 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(644): Laminas\ServiceManager\ServiceManager->createDelegatorFromName('Mezzio\\Applicat...', NULL)
#17 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(264): Laminas\ServiceManager\ServiceManager->doCreate('Mezzio\\Applicat...')
#18 /etc/shlink/bin/roadrunner-worker.php(19): Laminas\ServiceManager\ServiceManager->get('Mezzio\\Applicat...')
#19 /etc/shlink/bin/roadrunner-worker.php(32): {closure}()
#20 {main}

Next Doctrine\DBAL\Driver\PDO\Exception: SQLSTATE[HY000]: General error: 5 database is locked in /etc/shlink/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28
Stack trace:
#0 /etc/shlink/vendor/doctrine/dbal/src/Driver/PDO/Statement.php(123): Doctrine\DBAL\Driver\PDO\Exception::new(Object(PDOException))
#1 /etc/shlink/vendor/doctrine/dbal/src/Statement.php(190): Doctrine\DBAL\Driver\PDO\Statement->execute(NULL)
#2 /etc/shlink/vendor/doctrine/dbal/src/Statement.php(249): Doctrine\DBAL\Statement->execute(NULL)
#3 /etc/shlink/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php(284): Doctrine\DBAL\Statement->executeStatement()
#4 /etc/shlink/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php(1144): Doctrine\ORM\Persisters\Entity\BasicEntityPersister->executeInserts()
#5 /etc/shlink/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php(430): Doctrine\ORM\UnitOfWork->executeInserts(Object(Doctrine\ORM\Mapping\ClassMetadata))
#6 /etc/shlink/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php(403): Doctrine\ORM\UnitOfWork->commit(NULL)
#7 /etc/shlink/module/Rest/src/ApiKey/Repository/ApiKeyRepository.php(28): Doctrine\ORM\EntityManager->flush()
#8 /etc/shlink/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php(272): Shlinkio\Shlink\Rest\ApiKey\Repository\ApiKeyRepository->Shlinkio\Shlink\Rest\ApiKey\Repository\{closure}(Object(Doctrine\ORM\EntityManager))
#9 /etc/shlink/module/Rest/src/ApiKey/Repository/ApiKeyRepository.php(16): Doctrine\ORM\EntityManager->wrapInTransaction(Object(Closure))
#10 /etc/shlink/module/Rest/src/ApiKey/InitialApiKeyDelegator.php(29): Shlinkio\Shlink\Rest\ApiKey\Repository\ApiKeyRepository->createInitialApiKey('test_api_key')
#11 /etc/shlink/module/Rest/src/ApiKey/InitialApiKeyDelegator.php(19): Shlinkio\Shlink\Rest\ApiKey\InitialApiKeyDelegator->createInitialApiKey('test_api_key', Object(Laminas\ServiceManager\ServiceManager))
#12 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(620): Shlinkio\Shlink\Rest\ApiKey\InitialApiKeyDelegator->__invoke(Object(Laminas\ServiceManager\ServiceManager), 'Mezzio\\Applicat...', Object(Closure), NULL)
#13 /etc/shlink/vendor/mezzio/mezzio/src/Container/ApplicationConfigInjectionDelegator.php(55): Laminas\ServiceManager\ServiceManager::Laminas\ServiceManager\{closure}()
#14 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(620): Mezzio\Container\ApplicationConfigInjectionDelegator->__invoke(Object(Laminas\ServiceManager\ServiceManager), 'Mezzio\\Applicat...', Object(Closure), NULL)
#15 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(623): Laminas\ServiceManager\ServiceManager::Laminas\ServiceManager\{closure}()
#16 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(644): Laminas\ServiceManager\ServiceManager->createDelegatorFromName('Mezzio\\Applicat...', NULL)
#17 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(264): Laminas\ServiceManager\ServiceManager->doCreate('Mezzio\\Applicat...')
#18 /etc/shlink/bin/roadrunner-worker.php(19): Laminas\ServiceManager\ServiceManager->get('Mezzio\\Applicat...')
#19 /etc/shlink/bin/roadrunner-worker.php(32): {closure}()
#20 {main}

Next Doctrine\DBAL\Exception\LockWaitTimeoutException: An exception occurred while executing a query: SQLSTATE[HY000]: General error: 5 database is locked in /etc/shlink/vendor/doctrine/dbal/src/Driver/API/SQLite/ExceptionConverter.php:32
Stack trace:
#0 /etc/shlink/vendor/doctrine/dbal/src/Connection.php(1935): Doctrine\DBAL\Driver\API\SQLite\ExceptionConverter->convert(Object(Doctrine\DBAL\Driver\PDO\Exception), Object(Doctrine\DBAL\Query))
#1 /etc/shlink/vendor/doctrine/dbal/src/Connection.php(1878): Doctrine\DBAL\Connection->handleDriverException(Object(Doctrine\DBAL\Driver\PDO\Exception), Object(Doctrine\DBAL\Query))
#2 /etc/shlink/vendor/doctrine/dbal/src/Statement.php(194): Doctrine\DBAL\Connection->convertExceptionDuringQuery(Object(Doctrine\DBAL\Driver\PDO\Exception), 'INSERT INTO api...', Array, Array)
#3 /etc/shlink/vendor/doctrine/dbal/src/Statement.php(249): Doctrine\DBAL\Statement->execute(NULL)
#4 /etc/shlink/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php(284): Doctrine\DBAL\Statement->executeStatement()
#5 /etc/shlink/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php(1144): Doctrine\ORM\Persisters\Entity\BasicEntityPersister->executeInserts()
#6 /etc/shlink/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php(430): Doctrine\ORM\UnitOfWork->executeInserts(Object(Doctrine\ORM\Mapping\ClassMetadata))
#7 /etc/shlink/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php(403): Doctrine\ORM\UnitOfWork->commit(NULL)
#8 /etc/shlink/module/Rest/src/ApiKey/Repository/ApiKeyRepository.php(28): Doctrine\ORM\EntityManager->flush()
#9 /etc/shlink/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php(272): Shlinkio\Shlink\Rest\ApiKey\Repository\ApiKeyRepository->Shlinkio\Shlink\Rest\ApiKey\Repository\{closure}(Object(Doctrine\ORM\EntityManager))
#10 /etc/shlink/module/Rest/src/ApiKey/Repository/ApiKeyRepository.php(16): Doctrine\ORM\EntityManager->wrapInTransaction(Object(Closure))
#11 /etc/shlink/module/Rest/src/ApiKey/InitialApiKeyDelegator.php(29): Shlinkio\Shlink\Rest\ApiKey\Repository\ApiKeyRepository->createInitialApiKey('test_api_key')
#12 /etc/shlink/module/Rest/src/ApiKey/InitialApiKeyDelegator.php(19): Shlinkio\Shlink\Rest\ApiKey\InitialApiKeyDelegator->createInitialApiKey('test_api_key', Object(Laminas\ServiceManager\ServiceManager))
#13 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(620): Shlinkio\Shlink\Rest\ApiKey\InitialApiKeyDelegator->__invoke(Object(Laminas\ServiceManager\ServiceManager), 'Mezzio\\Applicat...', Object(Closure), NULL)
#14 /etc/shlink/vendor/mezzio/mezzio/src/Container/ApplicationConfigInjectionDelegator.php(55): Laminas\ServiceManager\ServiceManager::Laminas\ServiceManager\{closure}()
#15 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(620): Mezzio\Container\ApplicationConfigInjectionDelegator->__invoke(Object(Laminas\ServiceManager\ServiceManager), 'Mezzio\\Applicat...', Object(Closure), NULL)
#16 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(623): Laminas\ServiceManager\ServiceManager::Laminas\ServiceManager\{closure}()
#17 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(644): Laminas\ServiceManager\ServiceManager->createDelegatorFromName('Mezzio\\Applicat...', NULL)
#18 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(264): Laminas\ServiceManager\ServiceManager->doCreate('Mezzio\\Applicat...')
#19 /etc/shlink/bin/roadrunner-worker.php(19): Laminas\ServiceManager\ServiceManager->get('Mezzio\\Applicat...')
#20 /etc/shlink/bin/roadrunner-worker.php(32): {closure}()
#21 {main}

Next Laminas\ServiceManager\Exception\ServiceNotCreatedException: Service with name "Mezzio\Application" could not be created. Reason: An exception occurred while executing a query: SQLSTATE[HY000]: General error: 5 database is locked in /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php:649
Stack trace:
#0 /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php(264): Laminas\ServiceManager\ServiceManager->doCreate('Mezzio\\Applicat...')
#1 /etc/shlink/bin/roadrunner-worker.php(19): Laminas\ServiceManager\ServiceManager->get('Mezzio\\Applicat...')
#2 /etc/shlink/bin/roadrunner-worker.php(32): {closure}()
#3 {main}
  thrown in /etc/shlink/vendor/laminas/laminas-servicemanager/src/ServiceManager.php on line 649
```

### Expected behavior

Container starts with predefined `API_KEY`.
It works on version 3.5.4

### How to reproduce

```
docker run --rm -it \
    --name my_shlink \
    -p 8080:8080 \
    -e DEFAULT_DOMAIN="localhost" \
    -e IS_HTTPS_ENABLED=false \
    -e INITIAL_API_KEY=test_api_key \
    shlinkio/shlink:3.6.3
```

**Comment by acelaya** [comment]
That command worked for me:

```
{0:03}~ ➭ docker run --rm -it \
    --name my_shlink \
    -p 8080:8080 \
    -e DEFAULT_DOMAIN="localhost" \
    -e IS_HTTPS_ENABLED=false \
    -e INITIAL_API_KEY=test_api_key \
    shlinkio/shlink:3.6.3
Unable to find image 'shlinkio/shlink:3.6.3' locally
3.6.3: Pulling from shlinkio/shlink
f56be85fc22e: Already exists 
ace8de9a4ff5: Already exists 
ac818333da4c: Already exists 
10f4138fad9a: Already exists 
d0b6e626b009: Pull complete 
f2aff3f76714: Pull complete 
6bdfeb7eb0df: Pull complete 
fcc3efdbd672: Pull complete 
7390d2f1a204: Pull complete 
0401122727e6: Pull complete 
647d7ef0e223: Pull complete 
2528b606f9d0: Pull complete 
b9662ec8ee38: Pull complete 
09d3e755870a: Pull complete 
d86e72e15f24: Pull complete 
358fb8cab15f: Pull complete 
15f6704dca5f: Pull complete 
Digest: sha256:f46d7a4682f292d6c2ffa84c31c3c8a3be2a9c9b183b23a4049b69fbef1522a4
Status: Downloaded newer image for shlinkio/shlink:3.6.3
Initializing database if needed... Success!
Updating database... Success!
Generating proxies... Success!
Clearing entities cache... Success!
{"level":"debug","ts":1693605838521150818,"logger":"rpc","msg":"plugin was started","address":"tcp://127.0.0.1:6001","list of the plugins with RPC methods:":["resetter","informer","lock","app","jobs"]}
{"level":"debug","ts":1693605838521334145,"logger":"jobs","msg":"driver ready","pipeline":"shlink","driver":"memory","start":1693605838521229978,"elapsed":0.000103081}
{"level":"debug","ts":1693605838521348551,"logger":"memory","msg":"pipeline was started","driver":"memory","pipeline":"shlink","start":"2023-09-01 22:03:58.521346627 +0000 UTC","elapsed":"3.116µs"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 36, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 45, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 41, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 49, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 53, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 52, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 46, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 47, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 50, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 51, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 48, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 57, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 54, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 44, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 55, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 56, "internal_event_name": "EventWorkerConstruct"}
[INFO] RoadRunner server started; version: 2023.1.5, buildtime: 2023-06-08T14:45:05+0000
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 80, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 75, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 74, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 77, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 69, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 71, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 76, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 82, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 79, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 84, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 91, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 90, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 92, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 93, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 89, "internal_event_name": "EventWorkerConstruct"}
2023-09-01T22:03:58+0000	DEBUG	server      	worker is allocated	{"pid": 87, "internal_event_name": "EventWorkerConstruct"}
```

**Comment by bbednarek** [comment]
@acelaya it doesn't work for me. I have following setup locally. 
```
docker --version
Docker version 24.0.2, build cb74dfcd85
```

```
uname -m
x86_64
```
What setup do you have locally?

**Comment by acelaya** [comment]
Do you see this error only when providing `INITIAL_API_KEY`? Are you providing some other config but you have simplified the example here?

**Comment by acelaya** [comment]
Ok, docker version seems to have something to do. I can reproduce this on my laptop, with docker 24.0.5, but it works on my workstation with docker 20.10.17

**Comment by acelaya** [comment]
Fixing this was a bit tricky. See https://github.com/shlinkio/shlink/pull/1882#issue-1908397194 for details.

I will try to release v3.6.4 this afternoon if everything goes right.

**Comment by acelaya** [comment]
I have just released Shlink 3.6.4, and now this command works for me on both docker 20 and 24:

```bash
docker run --rm -it \
    --name my_shlink \
    -p 8080:8080 \
    -e DEFAULT_DOMAIN="localhost" \
    -e IS_HTTPS_ENABLED=false \
    -e INITIAL_API_KEY=test_api_key \
    shlinkio/shlink:3.6.4
```

Could you try and confirm?

**Comment by bbednarek** [comment]
Confirmed! It works for me as well!
Thanks for the quick fix 🚀
